File: ApolloCore/src/main/java/me/solarbam/apollo/apolloCore/utils/WorldUtils.java
package me.solarbam.apollo.apolloCore.utils;

import org.bukkit.entity.FallingBlock;

public class WorldUtils {
    private WorldUtils() {
        // Utility class
    }

    public static FallingBlock spawnFallingBlock(org.bukkit.Location location, org.bukkit.Material material) {
        org.bukkit.entity.FallingBlock fallingBlock = location.getWorld().spawnFallingBlock(location, material.createBlockData());
        fallingBlock.setDropItem(false);
        fallingBlock.setHurtEntities(false);
        return fallingBlock;
    }

}

File: ApolloCore/src/main/kotlin/me/solar/apollo/apolloCore/utils/WorldUtils.kt
package me.solar.apollo.apolloCore.utils

import org.bukkit.Location
import org.bukkit.Material
import org.bukkit.entity.FallingBlock

object WorldUtils {
    fun spawnFallingBlock(location: Location, material: Material): FallingBlock {
        val world = location.world
        val fallingBlock : FallingBlock? = world?.spawn(location, FallingBlock::class.java)
        if (fallingBlock == null) {
            throw IllegalStateException("No FallingBlock found")
        }
        fallingBlock.blockData.to(material.createBlockData())
        return fallingBlock
    }
}